src/hooks/useFitsScreen.ts

```diff
@@ -0,0 +1,183 @@
+    // element and the top edge of the screen.
+    fitsAbove: boolean;
+    // Whether the target element fits between the bottom edge of the reference
+    // element and the bottom edge of the screen.
+    fitsBelow: boolean;
+  };
+  fromLeft: {
+    // Whether the target element fits between the left edge of the reference
+    // element and the left edge of the screen.
+    fitsLeft: boolean;
+    // Whether the target element fits between the left edge of the reference
+    // element and the right edge of the screen.
+    fitsRight: boolean;
+  };
+  fromRight: {
+    // Whether the target element fits between the right edge of the reference
+    // element and the left edge of the screen.
+    fitsLeft: boolean;
+    // Whether the target element fits between the right edge of the reference
+    // element and the right edge of the screen.
+    fitsRight: boolean;
+  };
+  fromCenter: {
+    // Whether the target element fits between the horizontal center of the
+    // reference element and the left edge of the screen.
+    fitsLeft: boolean;
+    // Whether the target element fits between the horizontal center of the
+    // reference element and the right edge of the screen.
+    fitsRight: boolean;
+    // Whether the target element fits between the vertical center of the
+    // reference element and the top edge of the screen.
+    fitsAbove: boolean;
+    // Whether the target element fits between the vertical center of the
+    // reference element and the bottom edge of the screen.
+    fitsBelow: boolean;
+    fitsCentered: {
+      // Whether the top half of the target element fits between the vertical
+      // center of the reference element and the top edge of the screen.
+      fitsAbove: boolean;
+      // Whether the bottom half of the target element fits between the vertical
+      // center of the reference element and the bottom edge of the screen.
+      fitsBelow: boolean;
+      // Whether the left half of the target element fits between the horizontal
+      // center of the reference element and the left edge of the screen.
+      fitsLeft: boolean;
+      // Whether the right half of the target element fits between the
+      // horizontal center of the reference element and the right edge of
+      // the screen.
+      fitsRight: boolean;
+    };
+  };
+};
+
+/**
+ * A hook to determine if an element fits on the screen.
+ * @param targetNode The element to try and fit on the screen.
+ * @param referenceNode The element to use to position the target.
+ * @param callback The function to call when updating fitment info.
+ * @param spacer An additional space to leave between the target and reference.
+ * @param shouldCheck Whether the fitment info should be being checked.
+ * @param fromMouse Whether the target should be being positioned in relation
+ *                  to the mouse. In this case refernceNode will be used to
+ *                  listen for mouseover events.
+ */
+export const useFitsScreen = (
```

Comment:
This name implies a boolean return value.... maybe `useFitsWhereWithin` 🤷‍♂️

Reply:
How about `useWindowFitment`?

Reply:
Yeah that sounds good!